src/Razor/src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Microsoft.NET.Sdk.Razor.StaticWebAssets.targets

```diff
@@ -209,17 +219,56 @@ Copyright (c) .NET Foundation. All rights reserved.
        * Assets from the referenced packages. These will be implicitly included when nuget
          restores the package and includes the package props file for the package.
   -->
+
+    <!-- StaticWebAssets from the current project come from ResolveCurrentProjectStaticWebAssetsInputs which is a dependency of this target. -->
+    <!-- StaticWebAssets from referenced projects come from ResolveReferencedProjectsStaticWebAssets which is a dependency of this target. -->
+    <!-- StaticWebAssets from packages are already available, so we don't do anything. -->
   <Target
     Name="ResolveStaticWebAssetsInputs"
-    DependsOnTargets="$(ResolveStaticWebAssetsInputsDependsOn)">
+    DependsOnTargets="$(ResolveStaticWebAssetsInputsDependsOn)" />
 
-    <!-- StaticWebAssets from the current project -->
+  <!-- This is a helper task to compute the project references we need to invoke to retrieve
+       the static assets for a given application. We do it this way so that we can
+       pass additional build properties to compute the assets from the package when referenced
+       as a project. For example, Identity uses this hook to extend the project reference and
+       pass in the bootstrap version to use.
+  -->
+  <Target Name="_ResolveStaticWebAssetsProjectReferences"
+    DependsOnTargets="ResolveReferences">
 
-    <PropertyGroup>
-      <StaticWebAssetBasePath Condition="$(StaticWebAssetBasePath) == ''">_content/$(PackageId)</StaticWebAssetBasePath>
-    </PropertyGroup>
+    <ItemGroup>
+      <_StaticWebAssetsProjectReference
+        Include="%(ReferencePath.MSBuildSourceProjectFile)"
+        Condition="'%(ReferencePath.MSBuildSourceProjectFile)' != ''" />
+    </ItemGroup>
+
+  </Target>
+
+  <Target Name="ResolveReferencedProjectsStaticWebAssets" DependsOnTargets="$(ResolveReferencedProjectsStaticWebAssetsDependsOn)">
+
+    <MSBuild
+      Condition="'@(_StaticWebAssetsProjectReference->Count())' != '0'"
+      Projects="@(_StaticWebAssetsProjectReference)"
+      BuildInParallel="$(BuildInParallel)"
+      ContinueOnError="!$(BuildingProject)"
+      Targets="GetCurrentProjectStaticWebAssets"
+      SkipNonexistentTargets="true">
+      <Output TaskParameter="TargetOutputs" ItemName="_ReferencedProjectStaticWebAssets" />
+    </MSBuild>
 
     <ItemGroup>
+      <StaticWebAsset Include="@(_ReferencedProjectStaticWebAssets)" />
+    </ItemGroup>
+
+  </Target>
+
+  <Target Name="ResolveCurrentProjectStaticWebAssetsInputs" DependsOnTargets="$(ResolveCurrentProjectStaticWebAssetsInputsDependsOn)">
+
+      <PropertyGroup>
+        <StaticWebAssetBasePath Condition="$(StaticWebAssetBasePath) == ''">_content/$(PackageId)</StaticWebAssetBasePath>
+      </PropertyGroup>
+
+      <ItemGroup>
 
       <_ThisProjectStaticWebAsset
         Include="$(MSBuildProjectDirectory)\wwwroot\**"
```

Comment:
@rainersigwald will this globbing in target benefit from any caching or hit the disk over again!

Reply:
I'm not totally sure but I believe it'll hit the disk again.

Reply:
Is the suggestion here that we cache the results of this the first time we run it so that when it gets called in the child msbuild context doesn't go to disk?

Reply:
Wouldn't we have to go to disk to read the cache file either-way? (We would save evaluating the glob)

Reply:
It seems these would already have been read from disk during evaluation in the common case?

https://github.com/aspnet/AspNetCore-Tooling/blob/2e70e0cbd7d8b952313d1114cd750befca2b1453/src/Razor/src/Microsoft.NET.Sdk.Razor/Sdk/Sdk.Razor.StaticAssets.ProjectSystem.props#L30

I think that maybe this should return Content with certain metadata rather than re-scanning wwwroot\ but I may not yet fully understand the details or the scenario

Reply:
This is totally a separate issue if one at all so no need to block on this. I was just curious if this is expensive. Was trying to guess where the 3 remaining seconds might be. We can discuss on a separate issue.

Reply:
I think we did it this way because we couldn't figure out how to do something like `Include="@(Content)" Condition="%(Identity).StartsWith("wwwroot")"` or equivalent.

We essentially want anything that is Content but only that matches "wwwroot\**\*" as far as I know, there's no easy way to get that.

Reply:
Easy? No (microsoft/msbuild#1155). Possible? Yes, inside a target (as you are here):

```xml
      <Filtered Include="@(Content)" Condition="$([System.String]::Copy('%(Identity)').StartsWith('wwwroot'))" />
```

Reply:
Fixed it to be based on Content